datadog_checks_base/datadog_checks/base/checks/base.py

```diff
@@ -1084,15 +1085,21 @@ def _normalize_tags_type(self, tags, device_name=None, metric_name=None):
                     'Encoding error with device name `%r` for metric `%r`, ignoring tag', device_name, metric_name
                 )
 
+        disable_generic_tags = is_affirmative(self.instance.get('disable_generic_tags', False))
         for tag in tags:
             if tag is None:
                 continue
-
             try:
                 tag = to_native_string(tag)
             except UnicodeError:
                 self.log.warning('Encoding error with tag `%s` for metric `%s`, ignoring tag', tag, metric_name)
                 continue
+            tag_name, value = tag.split(':')
```

Comment:
I think we should put all of this behind the flag to avoid global performance degradation

Reply:
No, we want to emit the new tag always regardless of the feature flag. The flag is to stop sending the generic one.
The original idea was to add this to the query manager because it's a source of unknown tags, and fix the ones we know of on the affected integrations only like [mapr](https://github.com/DataDog/integrations-core/pull/9791).

The advantage of doing it here is that no further code changes are required for any integration, the disadvantage is a potential performance hit

Reply:
We could have another config option to enable the whole feature, leave it always hidden but set it only to true on the "offending integrations"